remoulade/actor.py

```
@@ -108,15 +98,23 @@ class Actor:
         and middleware.
     """
 
-    def __init__(self, fn, *, broker, actor_name, queue_name, priority, options):
+    def __init__(self, fn, *, actor_name, queue_name, priority, options):
         self.logger = get_logger(fn.__module__, actor_name)
         self.fn = fn
-        self.broker = broker
+        self.broker = None
         self.actor_name = actor_name
         self.queue_name = queue_name
         self.priority = priority
         self.options = options
-        self.broker.declare_actor(self)
+
+    def set_broker(self, broker):
+        invalid_options = set(self.options) - broker.actor_options
+        if invalid_options:
+            invalid_options_list = ", ".join(invalid_options)
+            message = "The following actor options are undefined: %s. " % invalid_options_list
+            message += "Did you forget to add a middleware to your Broker?"
+            raise ValueError(message)
+        self.broker = broker
```

Maybe raise an error in `send_with_options` if broker was not set?